controlplane/kubeadm/internal/controllers/status.go

```diff
@@ -135,3 +142,257 @@ func (r *KubeadmControlPlaneReconciler) updateStatus(ctx context.Context, contro
+		}
+	}
+
+	if kcp.Status.V1Beta2 == nil {
+		kcp.Status.V1Beta2 = &controlplanev1.KubeadmControlPlaneV1Beta2Status{}
+	}
+
+	kcp.Status.V1Beta2.ReadyReplicas = ptr.To(readyReplicas)
+	kcp.Status.V1Beta2.AvailableReplicas = ptr.To(availableReplicas)
+	kcp.Status.V1Beta2.UpToDateReplicas = ptr.To(upTpDateReplicas)
+}
+
+func setScalingUpCondition(_ context.Context, kcp *controlplanev1.KubeadmControlPlane, infrastructureObjectNotFound bool, preflightChecks internal.PreflightCheckResults) {
+	currentReplicas := kcp.Status.Replicas
+	desiredReplicas := ptr.Deref(kcp.Spec.Replicas, 0)
+	if !kcp.DeletionTimestamp.IsZero() {
+		desiredReplicas = 0
+	}
+
+	missingReferencesMessage := calculateMissingReferencesMessage(kcp, infrastructureObjectNotFound)
+
+	if currentReplicas >= desiredReplicas {
+		var message string
+		if missingReferencesMessage != "" {
+			message = fmt.Sprintf("ScalingUp can't happen %s", missingReferencesMessage)
+		}
+		v1beta2conditions.Set(kcp, metav1.Condition{
+			Type:    controlplanev1.KubeadmControlPlaneScalingUpV1Beta2Condition,
+			Status:  metav1.ConditionFalse,
+			Reason:  controlplanev1.KubeadmControlPlaneNotScalingUpV1Beta2Reason,
+			Message: message,
+		})
+		return
+	}
+
+	message := fmt.Sprintf("ScalingUp from %d to %d replicas", currentReplicas, desiredReplicas)
+	if missingReferencesMessage != "" {
+		message = fmt.Sprintf("%s is blocked %s", message, missingReferencesMessage)
+	}
+	messages := []string{message}
+
+	if preflightChecks.HasDeletingMachine {
+		messages = append(messages, "Waiting for Machine being deleted")
+	}
+
+	if preflightChecks.ControlPlaneComponentsNotHealthy {
+		messages = append(messages, "waiting for control plane components to be healthy")
+	}
+
+	if preflightChecks.EtcdClusterNotHealthy {
+		messages = append(messages, "waiting for etcd cluster to be healthy")
+	}
+
+	v1beta2conditions.Set(kcp, metav1.Condition{
+		Type:    controlplanev1.KubeadmControlPlaneScalingUpV1Beta2Condition,
+		Status:  metav1.ConditionTrue,
+		Reason:  controlplanev1.KubeadmControlPlaneScalingUpV1Beta2Reason,
+		Message: strings.Join(messages, "; "),
+	})
+}
+
+func setScalingDownCondition(_ context.Context, kcp *controlplanev1.KubeadmControlPlane, machines collections.Machines, preflightChecks internal.PreflightCheckResults) {
+	currentReplicas := kcp.Status.Replicas
+	desiredReplicas := ptr.Deref(kcp.Spec.Replicas, 0)
+	if !kcp.DeletionTimestamp.IsZero() {
+		desiredReplicas = 0
+	}
+
+	if currentReplicas <= desiredReplicas {
+		v1beta2conditions.Set(kcp, metav1.Condition{
+			Type:   controlplanev1.KubeadmControlPlaneScalingDownV1Beta2Condition,
+			Status: metav1.ConditionFalse,
+			Reason: controlplanev1.KubeadmControlPlaneNotScalingDownV1Beta2Reason,
+		})
+		return
+	}
+
+	messages := []string{fmt.Sprintf("ScalingDown from %d to %d replicas", currentReplicas, desiredReplicas)}
+	if preflightChecks.HasDeletingMachine {
+		messages = append(messages, "Waiting for Machine being deleted")
```

Comment:
```suggestion
		messages = append(messages, "waiting for Machine being deleted")
```
(same in l.278 + l.282)